docs/openmathreasoning1/dataset.md

```diff
@@ -158,17 +158,43 @@ consists of the following stages:
 You can run the full pipeline using [QwQ-32B](https://huggingface.co/Qwen/QwQ-32B) as solution generation model with
 
 ```
-python recipes/openmathreasoning/pipelines/solution_generation.py --mode full-qwq
+python recipes/openmathreasoning/pipelines/solution_generation.py --mode qwq
 ```
 
-You can specify a subset of stages using `--stages` argument and can switch between QwQ and R1 models using `--mode full-qwq` or `--mode full-r1`.
+You can specify a subset of stages using `--stages` argument and can switch between QwQ and R1 models using `--mode qwq` or `--mode r1`.
 
 If you want to run using [Nvidia NIM models](https://build.nvidia.com/models) on 10 example questions, add `--mode demo`.
 
 ## TIR solution generation pipeline
 
-Coming soon!
+[Tool-Integrated Reasoning (TIR) solution generation pipeline](https://github.com/NVIDIA/NeMo-Skills/tree/main/recipes/openmathreasoning/pipelines/solution_generation.py)
+focuses on generating solutions that leverage external tools, more specifically, a Python interpreter. This pipeline also consists of several stages:
 
-## GenSelect pipeline
+1.  [Generate solutions](../pipelines/generation.md) using a TIR-capable model (`generate_solutions` stage). These solutions interleave reasoning steps with executable code blocks.
+2.  [Fill majority answer](https://github.com/NVIDIA/NeMo-Skills/tree/main/nemo_skills/evaluation/aggregate_answers.py)
+    for problems without ground-truth answers (`fill_majority_answer` stage).
+3.  [Judge answers using an LLM](../pipelines/llm-as-a-judge.md), comparing the final answer to the ground-truth or majority answer (`judge_answers` stage).
+4.  Preprocess generations, including filtering and potentially standardizing code block formats (`preprocess_tir_generations` stage).
 
-Coming soon!
+## 
```

Comment:
I'm not sure this renders how you expect it. You can probably use 

```
!!! note
```

format or something similar. You can build the docs with `mkdocs serve` command to see how things look

Reply:
or just list them all together with [optional] or [stage0] in the beginning of those steps